### PR TITLE
Feature/map integration

### PR DESCRIPTION
**Pull request by grovecj**
Website supports creation of commodities and vehicles without parents. The resulting route is displayed on a map.  

**TODO**  
- [x] Send registration request to pathfinder-webserver  
- [x] Add .travis.yml  
- [x] Update README  
- [x] Add button to toggle from registration to login form  
- [x] Remove redundant self.map declaration (ExplorerController.js ln:9)  
- [x] Remove unused code (ExplorerController.js ln:57)

### CODE REVIEW

File: app/components/AppExplorer/ExplorerController.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
+(function() {
+    angular.module('pathfinder')
+        .controller('ExplorerController', ['uiGmapGoogleMapApi', '$scope', '$websocket', 'uiGmapIsReady', ExplorerController]);
+
+    function ExplorerController(uiGmapGoogleMapApi, $scope, $websocket, IsReady) {
+        var self = this;
+
+        // Public variables
+        self.map;
```
Comment (grovecj) [inline app/components/AppExplorer/ExplorerController.js]: redundant

File: app/components/AppExplorer/ExplorerController.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                            center: {
+                                latitude: pos.coords.latitude,
+                                longitude: pos.coords.longitude
+                            }
+                    });
+                });
+            });
+        }
+
+        uiGmapGoogleMapApi.then(function(maps) {
```
Comment (grovecj) [inline app/components/AppExplorer/ExplorerController.js]: Don't need this

File: e2e-tests/scenarios.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+/* https://github.com/angular/protractor/blob/master/docs/toc.md */
+
+describe('my app', function() {
```
Comment (grovecj) [inline e2e-tests/scenarios.js]: These are example tests provided by the angular seed app and should be removed

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+module.exports = function(config){
+  config.set({
+
+    basePath : './',
+
+    files : [
```
Comment (grovecj) [inline karma.conf.js]: This is the default karma.conf provided by the angular seed app. This will need to be modified to meet our project structure

**Comment by aj-michael** [comment]
The new routes code send recalculated routes every time a new vehicle or commodity is added. It looks like the current webclient is duplicating the routes everytime a new route is pushed.

[https://raw.githubusercontent.com/CSSE497/pathfinder/gh-pages/images/multiple%20vehicles.png](https://raw.githubusercontent.com/CSSE497/pathfinder/gh-pages/images/multiple%20vehicles.png)

**Comment by aj-michael** [comment]
On closer inspection, I believe the disconnected point may be the servers fault.

File: app/routes.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+                    templateUrl: 'components/landing/landing.html'
+                }).state('explorer', {
+                    url: '/app-explorer',
+                    templateUrl: 'components/AppExplorer/AppExplorer.html'
+                }).state('not-found', {
+                    url: '/not-found',
+                    templateUrl: '404.html'
+                });
+        });
+}());
```
Comment (robinsdm) [inline app/routes.js]: @cjgrove EOFs

**Comment by robinsdm** [comment]
Do the tests actually test anything?

**Comment by grovecj** [comment]
WOW! I WONDERED WHERE ALL THOSE EXCEPTIONS WERE COMING FROM -_- 

;^)

**Comment by grovecj** [comment]
the tests currently do nothing

**Comment by robinsdm** [comment]
Ok, so Travis failing isn't a big deal.

**Comment by grovecj** [comment]
Not at the moment. There is a bit of configuration work to to once we have some unit tests, but its fine for now

**Comment by robinsdm** [comment]
Should we merge it?

**Comment by grovecj** [comment]
I say go for it

**Comment by robinsdm** [comment]
Why is it going into master.

**Comment by grovecj** [comment]
because i wasn't paying attention

**Comment by grovecj** [comment]
I think i have to make a new pull request...

**Comment by robinsdm** [comment]
Yeah.

**Pull request closed without merging**